lib/puppet-syntax/hiera.rb

```diff
@@ -2,18 +2,44 @@
 
 module PuppetSyntax
   class Hiera
+
+    def check_hiera_key(key)
+      if key.is_a? Symbol
+        if key =~ /^:/
+          return "Puppet automatic lookup will not use leading '::'"
+        elsif key !~ /^[a-z]+$/ # we allow Hiera's own configuration
+          return "Puppet automatic lookup will not look up symbols"
+        end
+      elsif key !~ /^[a-z][a-z0-9_]+(::[a-z][a-z0-9_]+)*$/
+        if key =~ /[^:]:[^:]/
+          # be extra helpful
+          return "Looks like a missing colon"
+        else
+          return "Not a valid Puppet variable name for automatic lookup"
+        end
+      end
+    end
+
     def check(filelist)
       raise "Expected an array of files" unless filelist.is_a?(Array)
 
       errors = []
 
       filelist.each do |hiera_file|
         begin
-          YAML.load_file(hiera_file)
+          yamldata = YAML.load_file(hiera_file)
         rescue Exception => error
           errors << "ERROR: Failed to parse #{hiera_file}: #{error}"
+          next
+        end
+        yamldata.each do |k,v|
```

Comment:
It'd be a good idea to check the data type here before trying to iterate as loading an empty YAML file will return `false`.